pyxem/signals/diffraction_vectors.py

```diff
@@ -60,66 +60,51 @@
 """
 
 
-def _find_max_length_peaks(peaks):
-    """Worker function for generate_marker_inputs_from_peaks.
+def reverse_pos(peaks, ind=2):
+    """Reverses the position of the peaks in the signal."""
+    new_data = peaks[:, ind - 2 : ind]
+    return new_data[:, ::-1]
 
-    Parameters
-    ----------
-    peaks : :class:`~pyxem.diffraction_vectors.DiffractionVectors`
-        Identified peaks in a diffraction signal.
-
-    Returns
-    -------
-    longest_length : int
-        The length of the longest peak list.
 
-    """
-    length_of_longest_peaks_list = 0
-    with peaks.unfolded(unfold_navigation=True, unfold_signal=False):
-        for array in peaks.data:
-            if array.shape[0] > length_of_longest_peaks_list:
-                length_of_longest_peaks_list = array.shape[0]
-    return length_of_longest_peaks_list
+class ColumnSlicer:
+    def __init__(self, signal):
+        self.signal = signal
 
+    def __getitem__(self, item):
+        if isinstance(item, str):
+            item = self.signal.column_names.index(item)
+        elif isinstance(item, int):
+            pass
+        else:
+            raise ValueError("item must be a string or an int")
 
-def generate_marker_inputs_from_peaks(peaks):
-    """Takes a peaks (defined in 2D) object from a STEM (more than 1 image) scan
-    and returns markers.
-
-    Parameters
-    ----------
-    peaks : :class:`~pyxem.diffraction_vectors.DiffractionVectors`
-        Identifies peaks in a diffraction signal.
+        item = [
+            item,
+        ]
+        slic = self.signal.map(
+            lambda x, item: x[..., item], item=item, inplace=False, ragged=True
+        )
+        slic.offsets = self.signal.offsets[item[0]]
+        slic.scales = self.signal.scales[item[0]]
+        return slic
 
-    Example
-    -------
-    How to get these onto images::
 
-        mmx,mmy = generate_marker_inputs_from_peaks(found_peaks)
-        dp.plot(cmap='viridis')
-        for mx,my in zip(mmx,mmy):
-            m = hs.markers.point(x=mx,y=my,color='red',marker='x')
-            dp.add_marker(m,plot_marker=True,permanent=False)
+class BoolSlicer:
+    def __init__(self, signal):
+        self.signal = signal
 
-    """
-    max_peak_len = _find_max_length_peaks(peaks)
-    if peaks.data.dtype == np.dtype("O"):
-        navshape = peaks.data.shape
-    else:
-        navshape = peaks.axes_manager.navigation_shape[::-1]
-    pad = np.full((max_peak_len, *navshape, 2), np.nan)
-    for coordinate in np.ndindex(navshape):
-        array = peaks.data[coordinate]
-        sl = (slice(0, array.shape[0]), *coordinate, slice(None))
-        pad[sl] = array
-    x = pad[..., 0]
-    y = pad[..., 1]
-    return x, y
+    def __getitem__(self, item):
+        slic = self.signal.map(
+            lambda x, item: x[item], item=item, inplace=False, ragged=True
```

Comment:
Here you can also capture item in the lambda.

Reply:
Good catch!  That probably makes a bit more sense